common/src/core/java/xyz/wagyourtail/jsmacros/core/library/impl/FReflection.java

```diff
@@ -341,7 +344,66 @@ public Class<?> getClassFromClassBuilderResult(String cName) throws ClassNotFoun
     public LibraryBuilder createLibraryBuilder(String name, boolean perExec, String... acceptedLangs) throws NotFoundException, CannotCompileException {
         return new LibraryBuilder(name, perExec, acceptedLangs);
     }
+
+    /**
+     * To compile a java class, the game must be run with the java development kit (JDK) and not the
```

Comment:
honestly, I'm not sure about the addition of joor. especially with the jdk requirement. it'll be easier for users to install the groovy or kotlin addon at that point

Reply:
The text is a bit misleading, I probably have to update it, because I also tested the compile functionality without the jdk (it was still installed) and it worked. Thus, you may not even have to run the game with the jdk. 
The second reason I added joor is for reflection. Reflection is already hard to use in Java and much more annoying in guest languages and joor makes it really simple to access fields and invoke methods.
Now for the groovy / kotlin point. While I agree that these languages can probably provide the same functionality and I'm not even sure about this, it's just an addition to class builder. The library only is a few kilobyte and immensely improves the functionality provided by the class builder for those who use it, others are free to ignore it. The people using class builder or compile should not have any difficulies getting the jdk to work in case they need it. Following the same logic the class builder is also not needed, as you should be able to write the code in kotlin or groovy in a much easier way than with the builder.

Reply:
There is also a difference between writing classes and scripts. The kotlin and groovy extensions are not intended for writing Java classes to be used with Java, but rather for interacting with them. The class builder and compile on the other hand should not be used to write scripts, although certainly possible, but to create new classes.